Error during make: invalid application of ‘sizeof’ to incomplete type ‘boost::STATIC_ASSERTION_FAILURE<false>’

I'm compiling EOTCoin on Ubuntu 16.04. While building, make command `make -f makefile.unix` raises error as:
`rpcrawtransaction.cpp:242:77:   required from here
/usr/include/boost/variant/get.hpp:178:5: error: invalid application of ‘sizeof’ to incomplete type ‘boost::STATIC_ASSERTION_FAILURE<false>’
     BOOST_STATIC_ASSERT_MSG(
     ^
makefile.unix:187: recipe for target 'obj/rpcrawtransaction.o' failed:
make: *** [obj/rpcrawtransaction.o] Error 1`
Searching for the above error took me to [bitcoin repo](https://github.com/bitcoin/bitcoin/issues/6113#issuecomment-298234393) where the proposed fix is to change 

> rpcrawtransaction.cpp
at line# 242 from:
`const CScriptID& hash = boost::get<const CScriptID&>(address);`
to
`const CScriptID& hash = boost::get<CScriptID>(address);`

Applying the proposed fix does compile successfully, but i'm still wondering if this is a proper and safe way to make EOTCoin.

I get the same...
 ` required from here                                                                   
/usr/include/boost/variant/get.hpp:178:5: error: invalid application of ‘sizeof’ to incomplete type ‘boost::STATIC_ASSERT
ION_FAILURE<false>’                                                                                                      
     BOOST_STATIC_ASSERT_MSG(                                                                                            
     ^
Makefile:2431: die Regel für Ziel „build/rpcrawtransaction.o“ scheiterte     `

--- a/src/rpcrawtransaction.cpp
+++ b/src/rpcrawtransaction.cpp
@@ -239,7 +239,7 @@ Value listunspent(const Array& params, bool fHelp)
             CTxDestination address;
             if (ExtractDestination(pk, address))
             {
-                const CScriptID& hash = boost::get<const CScriptID&>(address);
+                const CScriptID& hash = boost::get<CScriptID>(address);
                 CScript redeemScript;
                 if (pwalletMain->GetCScript(hash, redeemScript))
                     entry.push_back(Pair("redeemScript", HexStr(redeemScript.begin(), redeemScript.end())));

